src/pform.ml

```diff
@@ -1,73 +1,93 @@
+module Var = struct
+  type t =
+    | Values of Value.t list
+    | Project_root
+    | First_dep
+    | Deps
+    | Targets
+    | Named_local
+end
+
+module Macro = struct
+  type t =
+    | Exe
+    | Dep
+    | Bin
+    | Lib
+    | Libexec
+    | Lib_available
+    | Version
+    | Read
+    | Read_strings
+    | Read_lines
+    | Path_no_dep
+    | Ocaml_config
+end
+
+module Expansion = struct
+  type t =
+    | Var   of Var.t
+    | Macro of Macro.t * string
+end
+
+type 'a t =
+  | No_info    of 'a
+  | Since      of 'a * Syntax.Version.t
+  | Deleted_in of 'a * Syntax.Version.t * string option
   | Renamed_in of Syntax.Version.t * string
 
+let values v                       = No_info (Var.Values v)
+let renamed_in ~new_name ~version  = Renamed_in (version, new_name)
+let deleted_in ~version ?repl kind = Deleted_in (kind, version, repl)
+let since ~version v               = Since (v, version)
+
 module Map = struct
-  type t = with_info String.Map.t
-
-  let values v                       = No_info (Values v)
-  let renamed_in ~new_name ~version  = Renamed_in (version, new_name)
-  let deleted_in ~version ?repl kind = Deleted_in (kind, version, repl)
-  let since ~version v               = Since (v, version)
-
-  let static =
-    let macro x = No_info x in
-    [ "targets", since ~version:(1, 0) Targets
-    ; "deps", since ~version:(1, 0) Deps
-    ; "project_root", since ~version:(1, 0) Project_root
-
-    ; "<", deleted_in First_dep ~version:(1, 0)
-             ~repl:"Use a named dependency instead:\
-                    \n\
-                    \n\  (deps (:x <dep>) ...)\
-                    \n\   ... %{x} ..."
-    ; "@", renamed_in ~version:(1, 0) ~new_name:"targets"
-    ; "^", renamed_in ~version:(1, 0) ~new_name:"deps"
-    ; "SCOPE_ROOT", renamed_in ~version:(1, 0) ~new_name:"project_root"
-
-    ; "exe", macro Exe
-    ; "bin", macro Bin
-    ; "lib", macro Lib
-    ; "libexec", macro Libexec
-    ; "lib-available", macro Lib_available
-    ; "version", macro Version
-    ; "read", macro Read
-    ; "read-lines", macro Read_lines
-    ; "read-strings", macro Read_strings
-
-    ; "dep", since ~version:(1, 0) Dep
-
-    ; "path", renamed_in ~version:(1, 0) ~new_name:"dep"
-    ; "findlib", renamed_in ~version:(1, 0) ~new_name:"lib"
-
-    ; "path-no-dep", deleted_in ~version:(1, 0) Path_no_dep
-    ; "ocaml-config", macro Ocaml_config
-    ]
-    |> String.Map.of_list_exn
+  type 'a map = 'a t String.Map.t
+
+  type t =
+    { vars   : Var.t   map
+    ; macros : Macro.t map
+    }
+
+  let static_vars =
+    String.Map.of_list_exn
+      [ "targets", since ~version:(1, 0) Var.Targets
+      ; "deps", since ~version:(1, 0) Var.Deps
+      ; "project_root", since ~version:(1, 0) Var.Project_root
+
+      ; "<", deleted_in Var.First_dep ~version:(1, 0)
+               ~repl:"Use a named dependency instead:\
+                      \n\
+                      \n\  (deps (:x <dep>) ...)\
```

Comment:
how is this escape interpreted by ocaml? It seems to me that in the other places we put `\n\` at the end and it does the right thing.

Reply:
Ah yes, this backslash is not necessary. Putting the `\n` at the begining is a trick to align the block.

Reply:
I updated the code